### DISCUSSION TITLE
Not able to build Theia from example

### DISCUSSION THREAD

**Discussion by kjellaso** [eclipse-theia/theia#14297] [Q&A] [answered]
Hi

I'm tyring to build theia from scratch following this example: https://theia-ide.org/docs/composing_applications,but the installation process fails.

When i set the yarn version to 2.4.3, i get the following installation error:
typescript@patch:typescript@npm%3A5.4.5#builtin<compat/typescript>::version=5.4.5&hash=8133ad: Cannot apply hunk #1 (set enableInlineHunks for details)

If i set yarn to the latest version, the installation works, but then the build fails with this error:
.yarn/__virtual__/@theia-application-manager-virtual-fccd3cd517/2/.yarn/berry/cache/@theia-application-manager-npm-1.54.0-3b599e5054-10c0.zip/node_modules/@theia/application-manager/node_modules/.bin/webpack: No such file or directory

I'm running on Fedora 40, Node versoin v20.17.0, Python 3.12.

Anyone seen this issue before?

-Andreas

**Comment by msujew** [comment, marked as answer]
Hey @kjellaso,

Theia requires Yarn v1 and Python <= 3.11 for everything to work out of the box. See also our [prerequisites](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites).

**Reply by kjellaso** [reply]
Thank you!

**Comment by srl295** [comment]
@msujew I have a question on this - why is application-manager directly sublaunching webpack and depending on its location? seems like it would be better to just [call `webpack()`](https://webpack.js.org/api/node/#webpack)

**Reply by msujew** [reply]
No idea, I didn't write the code ;)

I would assume it's legacy reasons - the function maybe didn't exist yet when the application manager was written (back in 2017).

**Reply by srl295** [reply]
@msujew would you be open to a PR to just call webpack directly? At least if the file it's looking for is missing

**Reply by srl295** [reply]
my specific issue is i'm trying to not require yarn/lerna (directly) but use npm workspaces, because that's what the parent project uses. This seems to be the only blocking issue at present. I got around it with a terrible local symlink that I'm not going to repeat here.

**Reply by msujew** [reply]
> would you be open to a PR to just call webpack directly? At least if the file it's looking for is missing

Sure, that makes sense.

> my specific issue is i'm trying to not require yarn/lerna (directly) but use npm workspaces, because that's what the parent project uses. This seems to be the only blocking issue at present. I got around it with a terrible local symlink that I'm not going to repeat here.

I'm am aware of at least one project (closed source though) that uses Theia with npm workspaces without too much trouble. It might be a setup issue.

**Reply by sdirix** [reply]
I encountered the same problem when trying to setup a simple Theia application with the latest `npm`. The webpack binary will not be linked to the `node_modules/@theia/application-manager/node_modules/bin` directory and the `theia` cli then fails.

I worked around this issue temporarily with [patch-package](https://github.com/ds300/patch-package), modifying the `application-manager` to also look into the parent directory to find the `webpack` binary, emulating the node resolving strategy.

I think at minimum we should implement this manual node resolving in the `theia` cli to enable more downstream architectures. We should also evaluate calling webpack directly as suggested above
